per/views.py

```diff
@@ -73,406 +37,300 @@ def handle_post(self, request, *args, **kwargs):
+        country_id = request.data.get('country_id', form.country_id)
+        ns = request.data.get('ns', form.ns)
+        ip = request.data.get('ip_address', form.ip_address)
+        comment = request.data.get('comment', form.comment)
+        validated = request.data.get('validated', form.validated)
+        finalized = request.data.get('finalized', form.finalized)
+        data = request.data.get('data', None)
+
+        # Update the Form properties and try to save
         try:
-            form = change_form(body)
-        except:
+            form.name = name
+            form.language = language
+            form.country_id = country_id
+            form.ns = ns
+            form.ip_address = ip
+            form.started_at = started_at
+            form.ended_at = ended_at
+            form.submitted_at = submitted_at
+            form.comment = comment
+            form.validated = validated
+            form.finalized = finalized
+            form.save()
+        except Exception:
             return bad_request('Could not change PER form record.')
 
-        if hasattr(form,'status_code') and form.status_code == 400:
-            return bad_request('Could not change PER form record due to inner failure.')
+        # Update Form Data of the Form
+        if data:
+            try:
+                with transaction.atomic():  # all or nothing
+                    for rubr in data:
+                        if rubr.id is None:
+                            raise Exception('PER Form Data ID was missing. Form ID: {}'.format(form_id))
 
-        if 'data' in body:
-            for rubr in body['data']:
-                rubr['form_id'] = body['id']
-                try:
-                    form_data = change_form_data(rubr, form)
-                except:
-                    return bad_request('Could not change PER formdata record.')
+                        form_data = FormData.objects.filter(form_id=form_id, question_id=rubr.id).first()
+                        if not form_data:
+                            raise Exception('Could not find PER Form Data record. \
+                                            Form ID: {} Form Data ID: {}'.format(form_id, rubr.id))
 
-                if hasattr(form_data,'status_code') and form_data.status_code == 400:
-                    return bad_request('Could not change PER form data record due to inner failure.')
+                        form_data.selected_option = rubr.op if rubr.op else form_data.selected_option
+                        form_data.notes = rubr.nt if rubr.nt else form_data.notes
 
-        return JsonResponse({'status': 'ok'})
+                        form_data.save()
+            except Exception as err:
+                return bad_request('Could not change PER formdata record. {}'.format(err))
 
+        return JsonResponse({'status': 'ok'})
 
-class DraftSent(PublicJsonPostView):
-    def handle_post(self, request, *args, **kwargs):
-        u = None
```

Comment:
@batpad @thenav56 I guessed this part (repeating in most Views below) is just a custom way for doing:
```python
authentication_classes = (authentication.TokenAuthentication,)
permissions_classes = (permissions.IsAuthenticated,)
```
So I just switched them out.